Vester/Private/Template/VesterTemplate.Tests.ps1

```diff
@@ -53,63 +85,57 @@ ForEach ($Test in $TestFiles) {
         }
     }
 
-    Describe -Name "$Scope Configuration: $TestName" -Fixture {
-        # Pull in $Title/$Description/$Desired/$Type/$Actual/$Fix from the test file
-        . $Test
+    If (($Final | Where-Object { $_.Scope -eq $Scope }).InventoryList -eq $null) {
+        Write-Verbose "No objects found in scope $Scope, skipping tests"
+        # Use continue to skip this test and go to the next loop iteration
+        continue
+    }    
 
-        # Pump the brakes if the config value is $null
-        If ($Desired -eq $null) {
-            Write-Verbose "Due to null config value, skipping test $TestName"
-            # Use continue to skip this test and go to the next loop iteration
-            continue
-        } Else {
-            $Datacenter = Get-Datacenter -Name $cfg.scope.datacenter -Server $cfg.vcenter.vc
-            # Use $Scope (parent folder) to get the correct objects to test against
-            # If changing values here, update the "$Scope -notmatch" test above as well
-            $InventoryList = switch ($Scope) {
-                'vCenter'    {$cfg.vcenter.vc}
-                'Datacenter' {$Datacenter}
-                'Cluster'    {$Datacenter | Get-Cluster -Name $cfg.scope.cluster}
-                'DSCluster'  {$Datacenter | Get-DatastoreCluster -Name $cfg.scope.dscluster}
-                'Host'       {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VMHost -Name $cfg.scope.host}
-                'VM'         {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VM -Name $cfg.scope.vm}
-                'Network'    {$Datacenter | Get-VDSwitch -Name $cfg.scope.vds}
-            }
-        } #If Desired
+    # Loops through each object in the inventory list for the specific scope
+    foreach($Object in $Inventory)
```

Comment:
I think that is the same point that jpsider points out below with his pictures. We should just need to swap the foreach loops. ill test it out and make a need commit.